tests/integration/targets/module_vault_kv2_write/tasks/cleanup.yml

```diff
@@ -0,0 +1,14 @@
+---
+- module_defaults:
+    vault_ci_kv2_destroy_all: "{{ vault_plugins_module_defaults_common }}"
+  block:
+  - name: Ensure test secrets are deleted
+    vault_ci_kv2_destroy_all:
+      path: "{{ item.path }}"
+      mount_point: "{{ item.mount | default(vault_kv2_mount_point) }}"
+    loop:
+      - path: "{{ vault_kv2_path }}/write1"
+      - path: "{{ vault_kv2_path }}/readonly"
+      - path: "{{ vault_kv2_path }}/writeonly"
+      - path: "write1"
+        mount: "{{ cas_required_vault_kv2_mount_point }}"
```

Comment:
I would prefer this step be moved to `setup.yml`, before the creation. The destroy should be idempotent so it shouldn't fail on non-existence, and this addresses a (very) minor edge case where a previous run failed without running the `always` block that invokes the cleanup, for whatever reason.

Mostly though, I'd just like all the steps to be together.

An example of another module that does this is `vault_kv2_delete`'s `setup.yml`.